lib/src/api/opt/endpoint/mod.rs

```diff
@@ -92,3 +102,49 @@ mod tests {
 		}
 	}
 }
+
+#[derive(Debug)]
+pub enum EndpointKind {
```

Comment:
Let's also avoid making this part of the public API for now.

Reply:
As we discussed, I have kept `EndpointKind` out of the public API in [3aa88f7](https://github.com/surrealdb/surrealdb/pull/2985/commits/3aa88f7cfd52825d06d4789f0fd919bfe740f359).